apis/kueue/v1beta1/clusterqueue_conversion_test.go

```diff
@@ -0,0 +1,477 @@
+				Spec: ClusterQueueSpec{
+					FlavorFungibility: &FlavorFungibility{
+						WhenCanPreempt: MayStopSearch,
+						WhenCanBorrow:  MayStopSearch,
+					},
+				},
+			},
+			expected: &v1beta2.ClusterQueue{
+				ObjectMeta: defaultObjectMeta,
+				Spec: v1beta2.ClusterQueueSpec{
+					FlavorFungibility: &v1beta2.FlavorFungibility{
+						WhenCanPreempt: v1beta2.MayStopSearch,
+						WhenCanBorrow:  v1beta2.MayStopSearch,
+					},
+				},
+			},
+		},
+		"FlavorFungibility TryNextFlavor preserved": {
+			input: &ClusterQueue{
+				ObjectMeta: defaultObjectMeta,
+				Spec: ClusterQueueSpec{
+					FlavorFungibility: &FlavorFungibility{
+						WhenCanPreempt: TryNextFlavor,
+						WhenCanBorrow:  TryNextFlavor,
+					},
+				},
+			},
+			expected: &v1beta2.ClusterQueue{
+				ObjectMeta: defaultObjectMeta,
+				Spec: v1beta2.ClusterQueueSpec{
+					FlavorFungibility: &v1beta2.FlavorFungibility{
+						WhenCanPreempt: v1beta2.TryNextFlavor,
+						WhenCanBorrow:  v1beta2.TryNextFlavor,
+					},
+				},
+			},
+		},
+		"complete ClusterQueue with all fields": {
+			input: &ClusterQueue{
+				ObjectMeta: defaultObjectMeta,
+				Spec: ClusterQueueSpec{
+					Cohort:          "prod-cohort",
+					AdmissionChecks: []AdmissionCheckReference{"multikueue-check"},
+					FlavorFungibility: &FlavorFungibility{
+						WhenCanPreempt: TryNextFlavor,
+						WhenCanBorrow:  MayStopSearch,
+					},
+					QueueingStrategy: StrictFIFO,
+				},
+				Status: ClusterQueueStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   "Active",
+							Status: metav1.ConditionTrue,
+							Reason: "Ready",
+						},
+					},
+					PendingWorkloads: 5,
+				},
+			},
+			expected: &v1beta2.ClusterQueue{
+				ObjectMeta: defaultObjectMeta,
+				Spec: v1beta2.ClusterQueueSpec{
+					CohortName: "prod-cohort",
+					AdmissionChecksStrategy: &v1beta2.AdmissionChecksStrategy{
+						AdmissionChecks: []v1beta2.AdmissionCheckStrategyRule{
+							{Name: "multikueue-check"},
+						},
+					},
+					FlavorFungibility: &v1beta2.FlavorFungibility{
+						WhenCanPreempt: v1beta2.TryNextFlavor,
+						WhenCanBorrow:  v1beta2.MayStopSearch,
+					},
+					QueueingStrategy: v1beta2.StrictFIFO,
+				},
+				Status: v1beta2.ClusterQueueStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   "Active",
+							Status: metav1.ConditionTrue,
+							Reason: "Ready",
+						},
+					},
+					PendingWorkloads: 5,
+				},
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			result := &v1beta2.ClusterQueue{}
+			if err := tc.input.ConvertTo(result); err != nil {
+				t.Fatalf("ConvertTo failed: %v", err)
+			}
+			if diff := cmp.Diff(tc.expected, result); diff != "" {
+				t.Errorf("unexpected conversion result (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestClusterQueueConvertFrom(t *testing.T) {
+	defaultObjectMeta := metav1.ObjectMeta{
+		Name: "test-clusterqueue",
+	}
+
+	testCases := map[string]struct {
```

Comment:
Please also add test case for "AdmissionChecksStrategy" used, demonstrating it stays in v1beta1 as AdmissionChecksStrategy